backend/typescript/services/implementations/reviewService.ts

```diff
@@ -189,7 +192,119 @@ class ReviewService implements IReviewService {
     };
   }
 
-  async deleteReview(id: string): Promise<void> {
+  // if deleteReview is called without parameter txn, it runs in a try catch block
```

Comment:
Instead of duplicating this code between the two cases (txn provided or not), could you maybe move this into a helper function? It'll make future changes here much easier